### ISSUE TITLE
Полиномиальная регрессия многомерная

### ISSUE THREAD

**Issue by klods2023** [klods2023/klods#7]
#библиотеки
!pip install pandas
!pip install scikit-learn
!pip install joblib

#Код полиномиальной регрессии
import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import PolynomialFeatures
from sklearn.metrics import r2_score, mean_absolute_error, mean_squared_error
import joblib
import matplotlib.pyplot as plt

#Загрузка данных из файла Excel
data = pd.read_excel('Data_base.xlsx')
X = data[['N', 'CH']]
y = data['Temp']

#Создание полиномиальных признаков
poly_features = PolynomialFeatures(degree=2)
X_poly = poly_features.fit_transform(X)

#Создание и обучение модели полиномиальной регрессии
model = LinearRegression()
model.fit(X_poly, y)

#Сохранение обученной модели
joblib.dump(model, 'polynomial_regression_model.pkl')
joblib.dump(poly_features, 'polynomial_features.pkl')

#Загрузка новых данных из файла Excel
new_data = pd.read_excel('New_data.xlsx')
new_data = new_data[['N', 'CH']]  # Удаление столбца 'Temp', если он не используется

#Преобразование новых данных в полиномиальные признаки
new_data_poly = poly_features.transform(new_data)

#Предсказание значений T для новых данных
predicted_T = model.predict(new_data_poly)

#Вывод предсказанных значений T
print('Предсказанные значения T для новых данных:')
for i, value in enumerate(predicted_T):
    print(f'Object {i+1}: {value}')

#Расчет коэффициента детерминации (R^2)
r2 = r2_score(y, model.predict(X_poly))
print(f'Коэффициент детерминации (R^2): {r2}')

#Расчет средней абсолютной ошибки (MAE)
mae = mean_absolute_error(y, model.predict(X_poly))
print(f'Средняя абсолютная ошибка (MAE): {mae}')

#Расчет среднеквадратичной ошибки (MSE)
mse = mean_squared_error(y, model.predict(X_poly))
print(f'Среднеквадратичная ошибка (MSE): {mse}')

#Отображение решения на графике
plt.scatter(X['N'], y, color='blue', label='Actual')
plt.scatter(new_data['N'], predicted_T, color='red', label='Predicted')
plt.xlabel('N')
plt.ylabel('Temp')
plt.legend()
plt.show()

**Comment by klods2023** [comment]
[Data_base.xlsx](https://github.com/klods2023/klods/files/13403794/Data_base.xlsx)
[New_data.xlsx](https://github.com/klods2023/klods/files/13403793/New_data.xlsx)



[Полиномиальная регрессия.docx](https://github.com/klods2023/klods/files/13403779/default.docx)
